pkg/authentication/handlers.go

```diff
@@ -189,7 +176,13 @@ func (a *API) checkState(r *http.Request, state string) error {
 	return nil
 }
 
-func badRequest(w http.ResponseWriter, err error) {
+func (a *API) badRequest(w http.ResponseWriter, err error) {
+	a.cookieManager.ClearIDTokenCookie(w)
+	a.cookieManager.ClearAccessTokenCookie(w)
+	a.cookieManager.ClearRefreshTokenCookie(w)
+	a.cookieManager.ClearNonceCookie(w)
+	a.cookieManager.ClearStateCookie(w)
```

Comment:
No you're not, what you depicted is correct. I understand this would be a little too "much", so for the handlers I'll remove the Cear*TokenCookie invocations.
In the middleware I would leave them since that's a different context.
Good catch!